### ISSUE TITLE
K3S Cluster Restore Failing: iptables

### ISSUE THREAD

**Issue by nileshp87** [k3s-io/k3s#9243]
Recently had a power outage that caused my server nodes to go down in a weird way. They had been running fine for months, and are all armv7 devices. 

I am now getting
```
Jan 16 03:21:38 hogsmeade2 k3s[1163068]: E0116 03:21:38.748135 1163068 proxier.go:1521] "Failed to execute iptables-restore" err=<
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         exit status 4: Warning: Extension recent revision 0 not supported, missing kernel module?
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         iptables-restore v1.8.8 (nf_tables):
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file or directory): rule in chain KUBE-SVC-NGSMBSSFR>
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file or directory): rule in chain KUBE-SVC-NGSMBSSFR>
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file or directory): rule in chain KUBE-SEP-AXIX24IP5>
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file or directory): rule in chain KUBE-SEP-YVRDWOSX6>
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file or directory): rule in chain KUBE-SVC-S4EMHPZ44>
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file or directory): rule in chain KUBE-SVC-S4EMHPZ44>
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file or directory): rule in chain KUBE-SEP-P5BU6RB7I>
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file or directory): rule in chain KUBE-SEP-EYR4KJ3M5>
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file or directory): rule in chain KUBE-SVC-NQL64TY2T>
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file or directory): rule in chain KUBE-SVC-NQL64TY2T>
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:         line 566: RULE_APPEND failed (No such file
Jan 16 03:21:38 hogsmeade2 k3s[1163068]:  >
```

When trying to start k3s.service. I am (and have been) using --prefer-bundled-bin. Toggling the value results in a different error:

```
Jan 16 03:18:08 hogsmeade2 k3s[1162501]: E0116 03:18:08.558628 1162501 proxier.go:1521] "Failed to execute iptables-restore" err=<
Jan 16 03:18:08 hogsmeade2 k3s[1162501]:         exit status 2: iptables-restore v1.8.7 (nf_tables): Couldn't load match `recent':No such f>
Jan 16 03:18:08 hogsmeade2 k3s[1162501]:         
Jan 16 03:18:08 hogsmeade2 k3s[1162501]:         Error occurred at line: 239
Jan 16 03:18:08 hogsmeade2 k3s[1162501]:         Try `iptables-restore -h' or 'iptables-restore --help' for more information.
Jan 16 03:18:08 hogsmeade2 k3s[1162501]:  >
```

Running k3s-uninstall, and newing up the cluster with the same flags results in a working install. However, running a restore against that, returns this issue.

I'm not really sure where to debug from here, but happy to provide any extra info I can.

**Environmental Info:**
K3s Version: 

Node(s) CPU architecture, OS, and Version: 

Cluster Configuration:

**Describe the bug:**

**Steps To Reproduce:**
Not sure, seems to be some combination of things. 

- Installed K3s: 
```
k3s version v1.28.5+k3s1 (5b2d1271)
go version go1.20.12
```

**Expected behavior:**
Would like to be able to restore the state of my old cluster, and let the rules re-create themselves? I'm not certain I understand how all this is working together though.

Expected: The agent services start successfully, and are externally accessible. 

**Actual behavior:**
k3s.service throws this error, and is not externally accessible. coredns is down.

**Additional context / logs:**

```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.3 LTS"
```

**Comment by tmathmeyer** [comment]
Lol

**Comment by brandond** [comment]
>`Warning: Extension recent revision 0 not supported, missing kernel module?`

You are missing kernel modules. You probably installed a kernel update package at some point, but hadn't rebooted. Make sure you have all the correct kernel module packages installed and loaded.

**Comment by NileshTheJoyful** [comment]
Do you have any clue as to:

1. What that module might be? (Or how to hunt down which one is missing)

Or 

2. Why a fresh install doesn't trigger this issue?

Googling around, I only find one reference to someone in an arch forum that was solved with a reboot. 

I've tried updating my system, and rebooted several times.

**Comment by NileshTheJoyful** [comment]
Do you have any clue as to:

1. What that module might be? (Or how to hunt down which one is missing)

Or 

2. Why a fresh install doesn't trigger this issue?

Googling around, I only find one reference to someone in an arch forum that was solved with a reboot. 

I've tried updating my system, and rebooted several times. 

Is there a doc where the requirements are listed out so I can compare them?

**Comment by nileshp87** [comment]
Do you have any clue as to:

1. What that module might be? (Or how to hunt down which one is missing)

Or 

2. Why a fresh install doesn't trigger this issue?

Googling around, I only find one reference to someone in an arch forum that was solved with a reboot. 

I've tried updating my system, and rebooted several times. 

Is there a doc where the requirements are listed out so I can compare them?

**Comment by nileshp87** [comment]
You were correct, it seems to have been a kernel problem. After a lot of head banging, and dealing with the vendor specific kernel build, I decided that it was just going to cause more headaches every kernel update. Especially since they had inconsistencies with which modules they were building version to version.

I opted to swap my x86_64 "agent" nodes to be dual purpose. Which should prevent weirdness like this in the future.

Thank you for your help.

Potential suggestion: update `k3s check-config` to  also check for the necessary host iptables modules. Would have saved me a lot of effort.

In case it helps anyone searching around, I was using a Khadas VIM4.
